synapseclient/client.py

```diff
@@ -3107,15 +3107,16 @@ def _create_table_snapshot(self, table, comment=None, label=None, activity=None)
         snapshot_body = {"snapshotComment": comment,
                          "snapshotLabel": label,
                          "snapshotActivityId": activity}
+        new_body = {key:value for key, value in snapshot_body.items() if value}
```

Comment:
`if value is not None` here since empty strings should probably be passed for these.